Assets/HoloToolkit/Utilities/Scripts/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        private void ConfigureTransformOverridePlane()
+        {
+            planePosition = TargetOverride.position;
+
+            Vector3 velocity = Vector3.zero;
+            if (TrackVelocity)
+            {
+                velocity = UpdateVelocity();
+            }
+            
+            // Place the plane at the desired depth in front of the camera and billboard it to the camera.
+            HolographicSettings.SetFocusPointForFrame(TargetOverride.position, -Camera.main.transform.forward, velocity);
+        }
+
+        /// <summary>
+        /// Configures the stabilization plane to update its position based on what your gaze intersects in the scene.
+        /// </summary>
+        private void ConfigureGazeManagerPlane()
+        {
+            Vector3 gazeOrigin = Camera.main.transform.position;
+            Vector3 gazeDirection = Camera.main.transform.forward;
+
+            // Calculate the delta between camera's position and current hit position.
+            float focusPointDistance = (gazeOrigin - GazeManager.Instance.Position).magnitude;
+            float lerpPower = focusPointDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                        : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, focusPointDistance, lerpPower * Time.deltaTime);
+
+            planePosition = gazeOrigin + (gazeDirection * currentPlaneDistance);
+
+            HolographicSettings.SetFocusPointForFrame(planePosition, -gazeDirection, Vector3.zero);
+        }
+
+        /// <summary>
+        /// Configures the stabilization plane to update based on a fixed distance away from you.
+        /// </summary>
+        private void ConfigureFixedDistancePlane()
+        {
+            float lerpPower = DefaultPlaneDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                          : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, DefaultPlaneDistance, lerpPower * Time.deltaTime);
+
+            planePosition = Camera.main.transform.position + (Camera.main.transform.forward * currentPlaneDistance);
+            HolographicSettings.SetFocusPointForFrame(planePosition, -Camera.main.transform.forward, Vector3.zero);
+        }
+
+        /// <summary>
+        /// Tracks the velocity of the target object to be used as a hint for the plane stabilization.
+        /// </summary>
+        private Vector3 UpdateVelocity()
+        {
+            Vector3 velocity = (TargetOverride.position - previousPosition) * FOCUS_POINT_FRAMERATE;
+            previousPosition = TargetOverride.position;
```

Comment:
Because velocity is only sometimes updated, and tracking can be controlled externally, there are cases where the previousPosition wouldn't represent last frames position, but instead the last frame that tracking was enabled. Maybe make TrackVelocity a property and reset previousPosition when going false->true.